src/server/string_family.cc

```diff
@@ -1313,23 +1313,20 @@ void StringFamily::MGet(CmdArgList args, Transaction* tx, SinkReplyBuilder* buil
     }
   }
 
-  SinkReplyBuilder::ReplyScope scope(builder);
   if (builder->GetProtocol() == Protocol::MEMCACHE) {
-    auto* rb = static_cast<MCReplyBuilder*>(builder);
     for (const auto& entry : res) {
       if (!entry)
         continue;
-      rb->SendValue(entry->key, entry->value, entry->mc_ver, entry->mc_flag);
+      builder->SendValue(entry->key, entry->value, entry->mc_ver, entry->mc_flag);
     }
-    rb->SendSimpleString("END");
+    builder->SendSimpleString("END");
   } else {
-    auto* rb = static_cast<RedisReplyBuilder*>(builder);
-    rb->StartArray(res.size());
+    builder->StartCollection(res.size(), SinkReplyBuilder::CollectionType::ARRAY);
     for (const auto& entry : res) {
       if (entry)
-        rb->SendBulkString(entry->value);
+        builder->SendBulkString(entry->value);
       else
-        rb->SendNull();
+        builder->SendNull();
```

Comment:
I will introduce a similar variant to `SendMGet` to avoid this in a follow up PR. For now I am fixing the UB.

Reply:
where is the UB? Why does it require changing the whole hierarchy?

Reply:
> Why does it require changing the whole hierarchy?

Because Vlad removed the common paths `SendMGetResponse` ...